src/main/java/com/ra4king/circuitsim/simulator/components/wiring/Clock.java

```diff
@@ -10,22 +11,77 @@
 import com.ra4king.circuitsim.simulator.Utils;
 import com.ra4king.circuitsim.simulator.WireValue;
 
+import javafx.beans.property.BooleanProperty;
+import javafx.beans.property.SimpleBooleanProperty;
+import javafx.beans.property.SimpleObjectProperty;
+import javafx.beans.value.ChangeListener;
+
 /**
  * @author Roi Atalla
  */
 public class Clock extends Component {
+	public static class EnabledInfo {
+		private final boolean enabled;
+		private final int hertz;
+		
+		public EnabledInfo(boolean enabled, int hertz) {
+			this.enabled = enabled;
+			this.hertz = hertz;
+		}
+		
+		public boolean getEnabled() {
+			return enabled;
+		}
+		
+		public int getHertz() {
+			return hertz;
+		}
+		
+		@Override
+		public boolean equals(Object o) {
+			if (o == this) {
+				return true;
+			}
+			
+			if (!(o instanceof EnabledInfo)) {
+				return false;
+			}
+			
+			EnabledInfo e = (EnabledInfo) o;
+			
+			return enabled == e.enabled && hertz == e.hertz;
+		}
+		
+		@Override
+		public int hashCode() {
+			return Objects.hash(enabled, hertz);
+		}
+	}
+	
 	private static class ClockInfo {
 		private final Map<Clock, Object> clocks = new ConcurrentHashMap<>();
 		private final Map<ClockChangeListener, Object> clockChangeListeners = new ConcurrentHashMap<>();
 		
 		private Thread currentClock;
+		private final SimpleObjectProperty<Clock.EnabledInfo> clockEnabled = new SimpleObjectProperty<>(
+				new Clock.EnabledInfo(false, 0));
 		private boolean clock;
 		
 		private long lastTickTime;
 		private long lastPrintTime;
 		private int tickCount;
 		private volatile int lastTickCount;
 		
+		private ClockInfo() {
+			clockEnabled.addListener((obs, oldValue, newValue) -> {
+				if(newValue.enabled) {
+					startClock(newValue.getHertz());
+				} else {
+					stopClock(false);
+				}
+			});
+		}
+		
 		void reset() {
 			stopClock();
```

Comment:
You missed this stopClock() call.

Reply:
Oh... That is quite embarrassing. I guess I made some changes after testing but I have no idea why I did that. Thank you for pointing that out.